src/transformers/models/speecht5/modeling_speecht5.py

```diff
@@ -2541,6 +2541,15 @@ def _generate_speech(
     vocoder: Optional[nn.Module] = None,
     output_cross_attentions: bool = False,
 ) -> Union[torch.FloatTensor, Tuple[torch.FloatTensor, torch.FloatTensor]]:
+    if speaker_embeddings is None:
+        logger.warning_once(
```

Comment:
Shouldn't this be an exception? A warning often gets un-noticed by the end-user.
In this case the user would just not be able to generate anything without a speaker_embedding.

IMO we should make it pretty clear with an `Exception`.

Reply:
Changing this code to require a speaker embedding would be a breaking change - it's possible that people are using fine-tuned versions of SpeechT5 without speaker embeddings, so throwing an error here would block them. I think a warning is sufficient

Reply:
I think it comes down to whether or not we recommend this. I agree with @Vaibhavs10, we are trying to move away from warnings everywhere. And I also agree with @sanchit-gandhi an error is breaking. 
But given the issue, I think it makes sense to either raise an error, or use a default embedding?

Reply:
Do you think we could instead register the default embedding as a [non-persistent buffer](https://pytorch.org/docs/stable/generated/torch.nn.Module.html#torch.nn.Module.register_buffer) (i.e. a parameter that is not saved to the state dict) and initialise it as one of the default speaker embeddings?

Reply:
I think it does yes!